var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -475,9 +475,7 @@ def find_optional_library(name, prefix):
 
         if '+petsc' in spec:
             options += [
-                'PETSC_OPT=%s' % spec['petsc'].headers.cpp_flags,
-                'PETSC_LIB=%s' %
-                ld_flags_from_library_list(spec['petsc'].libs)]
+                'PETSC_DIR=%s' % spec['petsc'].prefix]
```

Comment:
I think it is better to keep the previous set of options when `('^petsc+shared' in spec)` and otherwise use the option that sets `PETSC_DIR`. Setting `PETSC_DIR` will always link with all dependencies which is redundant when PETSc is build as shared library.